pkg/config/config.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package config
+
+import (
+	"bytes"
+	"io"
+	"strings"
+
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api"
+	"github.com/spf13/viper"
+	errors "golang.org/x/xerrors"
+)
+
+type options struct {
+	envPrefix    string
+	templatePath string
+}
+
+const (
+	cmdRoot = "ARIES"
+)
+
+// Option configures the package.
+type Option func(opts *options) error
+
+// FromReader loads configuration from in.
+// configType can be "json" or "yaml".
+func FromReader(in io.Reader, configType string, opts ...Option) api.ConfigProvider {
+	return func() (api.ConfigBackend, error) {
+		return initFromReader(in, configType, opts...)
+	}
+}
+
+// FromFile reads from named config file
+func FromFile(name string, opts ...Option) api.ConfigProvider {
+	return func() (api.ConfigBackend, error) {
+		backend, err := newBackend(opts...)
+		if err != nil {
+			return nil, err
+		}
+
+		if name == "" {
+			return nil, errors.New("filename is required")
+		}
+
+		// create new viper
+		backend.configViper.SetConfigFile(name)
+
+		// If a config file is found, read it in.
+		err = backend.configViper.MergeInConfig()
+		if err != nil {
+			return nil, errors.Errorf("loading config file failed: %w", err)
+		}
+
+		return backend, nil
+	}
+}
+
+// FromRaw will initialize the configs from a byte array
+func FromRaw(configBytes []byte, configType string, opts ...Option) api.ConfigProvider {
+	return func() (api.ConfigBackend, error) {
+		buf := bytes.NewBuffer(configBytes)
+		return initFromReader(buf, configType, opts...)
+	}
+}
+
+func initFromReader(in io.Reader, configType string, opts ...Option) (api.ConfigBackend, error) {
+	backend, err := newBackend(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	if configType == "" {
+		return nil, errors.New("empty config type")
+	}
+
+	// read config from bytes array, but must set ConfigType
+	// for viper to properly unmarshal the bytes array
+	backend.configViper.SetConfigType(configType)
+	err = backend.configViper.MergeConfig(in)
+	if err != nil {
+		return nil, err
+	}
+
+	return backend, nil
+}
+
+// WithEnvPrefix defines the prefix for environment variable overrides.
+// See viper SetEnvPrefix for more information.
```

Comment:
Suggest removing references to project dependencies in user-facing APIs. The godocs for user-facing APIs should explain themselves.

Reply:
done